extensions/2.0/Khronos/KHR_materials_clearcoat/README.md

```diff
@@ -0,0 +1,96 @@
+
+## Extending Materials
+
+The PBR clearcoat materials are defined by adding the `KHR_materials_clearcoat` extension to any compatible glTF material (excluding those listed above).  For example, the following defines a material like varnish using clearcoat parameters.
+
+```json
+{
+    "materials": [
+        {
+            "name": "varnish",
+            "extensions": {
+                "KHR_materials_clearcoat": {
+                    "clearcoatFactor": 1.0
+                }
+            }
+        }
+    ]
+}
+```
+
+### Clearcoat
+
+All implementations should use the same calculations for the BRDF inputs. Implementations of the BRDF itself can vary based on device performance and resource constraints. See [Appendix B](/specification/2.0/README.md#appendix-b-brdf-implementation) for more details on the BRDF calculations.
+
+|                                  | Type                                                                            | Description                            | Required             |
+|----------------------------------|---------------------------------------------------------------------------------|----------------------------------------|----------------------|
+|**clearcoatFactor**               | `number`                                                                        | The clearcoat layer intensity.         | No, default: `0.0`   |
+|**clearcoatTexture**              | [`textureInfo`](/specification/2.0/README.md#reference-textureInfo)             | The clearcoat layer intensity texture. | No                   |
+|**clearcoatRoughnessFactor**      | `number`                                                                        | The clearcoat layer roughness.         | No, default: `0.0`   |
+|**clearcoatRoughnessTexture**     | [`textureInfo`](/specification/2.0/README.md#reference-textureInfo)             | The clearcoat layer roughness texture. | No                   |
+|**clearcoatNormalTexture**        | [`normalTextureInfo`](/specification/2.0/README.md#reference-normaltextureinfo) | The clearcoat normal map texture.      | No                   |
+  
+The clearcoat formula `f_clearcoat` is computed using the specular term from the glTF 2.0 Metallic-Roughness material, defined in [Appendix B](/specification/2.0/README.md#appendix-b-brdf-implementation).  Use F0 equal `0.04`, base color white `1.0, 1.0, 1.0`, metallic value `0.0`, and the clearcoat roughness value defined in this extension as follows:
+
+```
+clearcoatRoughness = clearcoatRoughnessFactor * clearcoatRoughnessTexture.g
+```
+
+The following abstract code describes how the base and clearcoat layers should be blended together:
+  
+```
+clearcoatWeight = clearcoatTexture.r * clearcoatFactor * fresnel(0.04, NdotV)
+
+f = mix(f_emissive + f_diffuse + f_specular, f_clearcoat, clearcoatWeight)
```

Comment:
Hi Norbert,

Here's the original formula (note that "clearcoatBlendFactor" was renamed "clearcoatWeight" in this PR for clarity.  I'm open to renaming it back if needed.)

```
f = (f_emissive + f_diffuse) * (1.0 - clearcoatWeight) +
    mix(f_specular, f_clearcoat, clearcoatWeight)
```

But `mix` is not a black box, let's expand out what `mix` does, and separate emissive/diffuse too:

```
f = f_emissive * (1.0 - clearcoatWeight) +
    f_diffuse * (1.0 - clearcoatWeight) +
    f_specular * (1.0 - clearcoatWeight) +
    f_clearcoat * clearcoatWeight
```

From here we can see that the base layer's emissive, diffuse, and specular are all equally reduced by the clear coat.

```
f = (f_emissive + f_diffuse + f_specular) * (1.0 - clearcoatWeight) +
    f_clearcoat * clearcoatWeight
```

By now it should be clear that `mix` can handle all of this in one go.

```
f = mix(f_emissive + f_diffuse + f_specular, f_clearcoat, clearcoatWeight)
```

This is the same formula, just optimized a bit.  It yields numerically equivalent answers.

Reply:
lol ... yes u are right